### ISSUE TITLE
ClassCastException with Queue in Scala 2.12.2

### ISSUE THREAD

**Issue by etorreborre** [scala/bug#10298]
This works fine with Scala `2.12.1`
```scala
import collection.immutable._
val q = Queue.empty[Int]
val v = Vector.empty[Int]

// ClassCastException
q ++ v
```

**Comment by edmundnoble** [comment]
My bad, this was the most controversial part of the code I added and we all kind of blanked. The fix is just a `to[Queue]` call inside `Queue.++`.

**Comment by edmundnoble** [comment]
PR at https://github.com/scala/scala/pull/5891.

**Comment by SethTisue** [comment]
> Anyone know how/when this can get fixed

in 2.12.3, but note that your fix is already available for testing as Scala 2.12.3-bin-105e2c1-SNAPSHOT, and once it's merged it'll be also available in some nightly-build Scala 2.12.3-bin-abcd123 (for a SHA to be named later)
